### PR TITLE
105-interface-for-sequencer-to-ingest-stream-of-streams-of-otelevent

### PR DESCRIPTION
**Pull request by AdrianMontaguSmartDCSIT**
This PR adds an interface within the otel sequencer module that takes in a generator of generator of OTelEvents and returns a generator of generator of PVEvents.

### CODE REVIEW

File: tel2puml/sequence_otel_v2.py [hunk 1]
```diff
@@ -257,3 +255,54 @@ def sequence_otel_jobs(
...
+    :param event_to_async_group_map: A dictionary mapping event types to
+    groups of events that occur asynchronously, defaults to None.
+    :type event_to_async_group_map: `dict`[`str`, `dict`[`str`, `str`]] |
+    `None`
+    :return: A generator of jobs (generators) of PVEvents.
+    :rtype: `Generator`[`Generator`[:class:`PVEvent`, `Any`, `None`], `Any`,
+    `None`]
+    """
+
+    def job_ids_to_eventid_to_otelevent_map(
```
Comment (FreddieMatherSmartDCSIT) [inline tel2puml/sequence_otel_v2.py:284]: shouldn't be nested in another function. just extract out and then test this separately

File: tel2puml/sequence_otel_v2.py [hunk 2]
```diff
@@ -257,3 +255,54 @@ def sequence_otel_jobs(
...
+        :param job_id_streams: A generator of generators, where each inner
+        generator yields OTelEvent objects grouped by job ID.
+        :type job_id_streams: `Generator`[`Generator`[:class:`OTelEvent`,
+        `Any`, `None`], `Any`, `None`]
+        :return: A generator of dictionaries mapping event IDs to OTelEvent
+        objects for each job.
+        :rtype: `Generator`[`dict`[`str`, :class:`OTelEvent`], `Any`, `None`]
+        """
+        for job_group in job_id_streams:
+            event_id_map = {}
```
Comment (FreddieMatherSmartDCSIT) [inline tel2puml/sequence_otel_v2.py:299]: just a preference here - might as well use dict comprehension

**Review by FreddieMatherSmartDCSIT — CHANGES_REQUESTED** [review]
Just need to extract that function out of the scope of the other function and test - should be a very simple test. Just for future, my preference is not to have functions within the scope of another function unless you're doing a closure or testing

File: tel2puml/sequence_otel_v2.py (hunk 1 above)
Comment (AdrianMontaguSmartDCSIT) [reply tel2puml/sequence_otel_v2.py:284]: separated it out

**Review by FreddieMatherSmartDCSIT — APPROVED** [review]
Looks good to me

**Pull request merged**